src/alexandra_ai_eval/task.py

```diff
@@ -260,28 +261,35 @@ def _evaluate_single_iteration(
             torch.manual_seed(703 + idx)
             torch.cuda.manual_seed_all(703 + idx)
 
+            # Start carbon emissions tracking
+            if self.evaluation_config.track_carbon_emissions:
+                self.carbon_tracker.start()
+
             # Define batch size, which depends on whether we are testing or not
             batch_size = (
                 2
                 if self.evaluation_config.testing
-                else 4
-                if self.task_config.name == "automatic-speech-recognition"
                 else 32
+                if self.task_config.name == "automatic-speech-recognition"
+                else 256
             )
-
-            # Start carbon emissions tracking
-            if self.evaluation_config.track_carbon_emissions:
-                self.carbon_tracker.start()
-
-            # Get model predictions
-            model_predictions = self._get_model_predictions(
-                model=model,
-                tokenizer=tokenizer,
-                processor=processor,
-                prepared_dataset=prepared_dataset,
-                batch_size=batch_size,
-                framework=framework,
-            )
+            while batch_size > 1:
+                try:
+                    # Get model predictions
+                    model_predictions = self._get_model_predictions(
+                        model=model,
+                        tokenizer=tokenizer,
+                        processor=processor,
+                        prepared_dataset=prepared_dataset,
+                        batch_size=batch_size,
+                        framework=framework,
+                    )
+                except (OutOfMemoryError, RuntimeError) as e:  # type: ignore
+                    # If we encounter an OOM error, or Runtimeerror in case of CPU, we halve the batch size
+                    # and try again, we raise an error in case the batch size is reduced from 1 to 0.
+                    batch_size //= 2
+                    if batch_size == 0:
+                        raise e
```

Comment:
I think we need to clear CUDA cache here as well, as we get a memory leak otherwise?

Reply:
Yes, of course! :)